### PR TITLE
Add ignore rules comparing actual and expected values

### PR DESCRIPTION
**Pull request by kaituo**
### Description

This PR introduces suppression anomaly rules under the Advanced Settings section, enabling users to suppress anomalies based on the difference between expected and actual values, either as an absolute value or a relative percentage.

Testing:
* Added unit tests to verify the suppression rules functionality.
* Conducted manual end-to-end (e2e) tests to validate the implementation.

Note most of changes are snapshot file changes:
```
git diff 3ed0058a9c8b52dc8a34b11f4869f0dda6904e2a --stat -- '*.snap'
 .../containers/__tests__/__snapshots__/DetectorConfig.test.tsx.snap | 1563 +++++++++++++++++++++++++++++-
 .../__tests__/__snapshots__/AdditionalSettings.test.tsx.snap        |  824 ++++++++++------
 .../__tests__/__snapshots__/ModelConfigurationFields.test.tsx.snap  |  413 +++++---
 .../__tests__/__snapshots__/ReviewAndCreate.test.tsx.snap           |  802 +++++++++------
 4 files changed, 2814 insertions(+), 788 deletions(-)
```

demo:

https://github.com/user-attachments/assets/612dce9a-d7c4-4b94-aed7-4b42465ece7f

### Check List

- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
- [X] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kaituo** [comment]
link check error. Not sure why it happens as https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/commit/c1da0e52eb0bcb3beee23f642686661da634f7f4 is accessible. 

### Errors in release-notes/opendistro-for-elasticsearch.anomaly-detection-kibana-plugin.release-notes-1.7.0.0.md

* [406] [https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/commit/c1da0e52eb0bcb3beee23f642686661da634f7f4](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/commit/c1da0e52eb0bcb3beee23f642686661da634f7f4) | Failed: Network error: Not Acceptable

**Comment by kaituo** [comment]
"Run binary installation" failed due to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7919

File: public/pages/ConfigureModel/utils/helpers.ts [hunk 1]
```diff
@@ -425,3 +438,138 @@ export const getCustomValueStrArray = (imputationMethodStr : string, detector: D
...
+    // Convert each condition to a readable string
+    return rule.conditions.map((condition) => {
+      const featureName = condition.featureName;
+      const thresholdType = condition.thresholdType;
+      let value = condition.value;
+      const isPercentage = thresholdType === ThresholdType.ACTUAL_OVER_EXPECTED_RATIO || thresholdType === ThresholdType.EXPECTED_OVER_ACTUAL_RATIO;
+
+      // If it is a percentage, multiply by 100
+      if (isPercentage) {
+        value *= 100;
```
Comment (jackiehanyang) [inline public/pages/ConfigureModel/utils/helpers.ts:457]: nit: could value be undefined or null here? Should we handle it here? something like `value = (value ?? 0) * 100;`

**Review by jackiehanyang — APPROVED** [review]

File: public/pages/ConfigureModel/utils/helpers.ts (hunk 1 above)
Comment (kaituo) [reply public/pages/ConfigureModel/utils/helpers.ts:457]: I made sure the value is a positive number before saving it in detector: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/1708b0b16684f28079cc36fa964b9e405ef0dddc/public/pages/ConfigureModel/utils/helpers.ts#L506-L518

On the backend, the default value is 0. If relative value is there, it has to be a double. So it cannot be undefined or null.

**Pull request merged**